internal/validator/cosignvalidator/cosign_validator.go

```diff
@@ -35,10 +34,16 @@ type CosignValidator struct {
 	Name string `validate:"required"`
 	// type of the validator (will always be "cosign")
 	Type string `validate:"eq=cosign"`
-	// url of the Rekor instance
-	Rekor string `validate:"url"`
+	// Rekor instance
+	Rekor *client.Rekor
+	// public key of the Rekor instance
+	RekorPubkey *cosign.TrustedTransparencyLogPubKeys
+	// certificate of the fulcio instance
+	FulcioCert *x509.CertPool
+	// public key of the CT log
+	CTLogPubkey *cosign.TrustedTransparencyLogPubKeys
```

Comment:
You only ever use this in a private function, so it should be private. Also, the description isn't true, see above.